modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/introspection/validation/MetadataComponentModelValidatorTestCase.java

```diff
@@ -0,0 +1,131 @@
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@SmallTest
+@RunWith(MockitoJUnitRunner.class)
+public class MetadataComponentModelValidatorTestCase extends AbstractMuleTestCase
+{
+
+    @Mock(answer = RETURNS_DEEP_STUBS)
+    private ExtensionModel extensionModel;
+
+    @Mock
+    private RuntimeOperationModel operationModel;
+
+    @Mock
+    private RuntimeSourceModel sourceModel;
+
+    private MetadataComponentModelValidator validator = new MetadataComponentModelValidator();
+
+    private class SimplePOJO
+    {
+
+    }
+
+    public static class SimpleOutputResolver implements MetadataOutputResolver
+    {
+        @Override
+        public MetadataType getOutputMetadata(MetadataContext context, MetadataKey key) throws MetadataResolvingException, ConnectionException
+        {
+            return null;
+        }
+
+    }
+
+    @Before
+    public void before()
+    {
+        when(extensionModel.getOperationModels()).thenReturn(asList(operationModel));
+        when(extensionModel.getSourceModels()).thenReturn(asList(sourceModel));
+
+        when(operationModel.getReturnType()).thenReturn(toMetadataType(String.class));
+        when(operationModel.getName()).thenReturn("operation");
+        when(operationModel.getMetadataResolverFactory()).thenReturn(new NullMetadataResolverFactory());
+
+        when(sourceModel.getReturnType()).thenReturn(toMetadataType(String.class));
+        when(sourceModel.getName()).thenReturn("source");
+        when(sourceModel.getMetadataResolverFactory()).thenReturn(new NullMetadataResolverFactory());
+    }
+
+    @Test
+    public void valid()
+    {
+        validator.validate(extensionModel);
+    }
+
+    @Test(expected = IllegalModelDefinitionException.class)
+    public void operationObjectReturnType()
+    {
+        when(operationModel.getReturnType()).thenReturn(toMetadataType(Object.class));
+        validator.validate(extensionModel);
+    }
+
+    @Test(expected = IllegalModelDefinitionException.class)
+    public void operationMapReturnType()
+    {
+        when(operationModel.getReturnType()).thenReturn(toMetadataType(Map.class));
+        validator.validate(extensionModel);
+    }
+
+    @Test(expected = IllegalModelDefinitionException.class)
+    public void sourceObjectReturnType()
+    {
+        when(sourceModel.getReturnType()).thenReturn(toMetadataType(Object.class));
+        validator.validate(extensionModel);
+    }
+
+    @Test(expected = IllegalModelDefinitionException.class)
+    public void sourceMapReturnType()
+    {
+        when(sourceModel.getReturnType()).thenReturn(toMetadataType(Map.class));
+        validator.validate(extensionModel);
+    }
+
+    @Test
+    public void sourcePOJOReturnType()
+    {
+        when(sourceModel.getReturnType()).thenReturn(toMetadataType(SimplePOJO.class));
+        validator.validate(extensionModel);
+    }
+
+    @Test
+    public void sourceObjectReturnTypeWithOutputResolver()
```

Comment:
the test names is the complete opposite of what you're doing which is providing a non null output resolver